akka-http/src/main/scala/com/avast/grpc/jsonbridge/akkahttp/AkkaHttp.scala

```diff
@@ -41,38 +42,87 @@ object AkkaHttp extends SprayJsonSupport with DefaultJsonProtocol {
       .map(_ / Segment / Segment)
       .getOrElse(Segment / Segment)
 
+    logger.info(s"Creating Akka HTTP service proxying gRPC services: ${bridge.servicesNames.mkString("[", ", ", "]")}")
+
     post {
       path(pathPattern) { (serviceName, methodName) =>
-        extractRequest { req =>
-          req.header[`Content-Type`] match {
+        extractRequest { request =>
+          val headers = request.headers
+          request.header[`Content-Type`] match {
             case Some(`JsonContentType`) =>
-              entity(as[String]) { json =>
-                val methodCall = bridge
-                  .invoke(GrpcMethodName(serviceName, methodName), json, mapHeaders(req.headers))
-                  .toIO
-                  .unsafeToFuture()
+              entity(as[String]) { body =>
+                val methodName0 = GrpcMethodName(serviceName, methodName)
+                val headers0 = mapHeaders(headers)
+                val methodCall = bridge.invoke(methodName0, body, headers0).toIO.unsafeToFuture()
                 onComplete(methodCall) {
-                  case Success(Right(r)) =>
+                  case Success(result) =>
+                    result match {
+                      case Right(resp) =>
+                        logger.trace("Request successful: {}", resp.substring(0, 100))
+                        respondWithHeader(JsonContentType) {
+                          complete(resp)
+                        }
+                      case Left(er) =>
+                        er match {
+                          case BridgeError.GrpcMethodNotFound =>
+                            val message = s"Method '${methodName0.fullName}' not found"
+                            logger.warn(message)
```

Comment:
commenting about all related issue levels: I know we talked about this -- I'm changing this based on my recent experience

just to make it clear, I don't use it for developing my services. I need this for operations in production. Not having specifically this issue on line logged was one of the contributing factors why Chrlic wasn't releasing VPSes for a considerably long time -- that's a huge issue, AFAIK

I believe this is a better default in general, not just for my usecase. The responsibility of this component (i.e. `grpc-json-bridge`) is to translate between HTTP/JSON and gRPC, if non-existent method is called, it's an issue. I know you've said that it's the client's responsibility to handle such issue, but not all clients are perfect (like Chrlic, it's a host stinking Perl/Docker piece of shit, and I can't fix it in a day).

Logging more is not a big problem, in Kibana, it's trivial to filter out the lines you don't care about. But it's impossible to add missing logs back after the fact.

Reply:
Thinking about this for a while... maybe you're right, that `warn` for such a case would be maybe too hysterical of `grpc-json-bridge`.
I changed it to `info` (this "Method not found" and the JSON parsing error).
On the other hand, I've raised the `"Unknown request error"` to `warn` because that's an error that really shouldn't happen, we don't know anything how or why it happened and most likely comes from the internals of `grpc-json-bridge` itself.

is that something you could agree with?

Reply:
to explain my philosophy about log levels: for example "gRPC request error" doesn't happen here (as in `grpc-json-bridge`), but in another component, so I log it just as `debug`

Reply:
I agree with having `"Unknown request error"` if that's a case it really shouldn't happen, because that indicates an error in the service.  

I also believe that this has already caused you some troubles, it's a real situation and it probably didn't happen for the last time.  

However, I still strongly believe it's a responsibility of the client! Imagine situation that the client uses some 3rd party service (e.g. Google) - then you don't have any service logs, just response code and payload. You should think about this issue the same way - the service must not be responsible for errors (predictable) of any client. And from the other side, if the client is not able to detect WTH happened wrong with the request, it doesn't deserve to exist :smiley:   

_From my POV:_ The client must indicate something is wrong and it should, if possible, output the error into logs. From that, you'd be able to recognize what's going wrong or, at least, which service is failing. Then you can take advantage of the logs, **lower the logging threshold for the service**, run the client again and see exactly what happens.

_Now back to your situation._ How do you recognize the client is not working? How do you recognize the situation, when you even open the service logs and try to find the problem here? And when you do that, why is it a problem to adjust the logging level and run the client again?

Reply:
This is all a nice theory, and I kinda agree, that's I've made the compromise to lower it to `info`. But we need to be pragmatic: in practice, logging malformed input as `info` (not `error` of course) is good default, because it's useful.

I want to emphasize again, that not having this in logs, caused a serious disruption (or more precisely, made the disruption take longer to fix) to Avast core business.

You can say that shitty clients don't deserve to exist, but my service *has* to work with Chrlic. And even though Chrlic might not be the best designed system or the easiest to operate and debug, it *must* work, otherwise we can close the shop.

Reply:
I'm for warn level but info is also OK for me. The reason is that I believe that as backend engineer, I could want to know that some client is calling my services in a wrong way. E.g. after a deployment that could potentially brake the API.

I also believe that who should be alerted on the first place is the engineering team responsible for the client.

Reply:
You should detect there is some bug after deployment with a alert based on metrics, not by checking logs.  

Ondro, it's not about your service has to work with the client; the client has to work with the service. I don't want to repeat myself neither.

I had a discussion about this with @petr-k and @luminaar, both kind of share thew view with me.

Reply:
I'm in @jendakol's camp.

I strongly believe that whenever the client does something wrong, it should be guided by good, descriptive error response. E.g., not getting just a plain `400`, but also a message explaining what exactly was wrong with how the request was made. This stipulation being satisfied, I see no use logging this server-side, except for debugging the server. Hence my proposal also being the `DEBUG` log level. Should you get yourself into the unfortunate place where you _do_ need to debug it, you just reconfigure your logging. That's what log levels are for - not just to carry the information about message severity, but also to enable you to see more detail by "flipping a switch" should need be.